main.go

```diff
@@ -214,34 +210,47 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, cCache *cache.Cache
 		os.Exit(1)
 	}
 
-	crds := findCustomResources(ctx, mgr)
-
 	opts := []jobframework.Option{
 		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
 		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 		jobframework.WithKubeServerVersion(serverVersionFetcher),
 	}
 	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
 		log := setupLog.WithValues("jobFrameworkName", name)
-		if isFrameworkEnabled(cfg, name) && crds.Has(name) {
-			if err := cb.NewReconciler(
-				mgr.GetClient(),
-				mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, constants.KueueName)),
-				opts...,
-			).SetupWithManager(mgr); err != nil {
-				log.Error(err, "unable to create controller")
-				return err
-			}
-			if err := cb.SetupWebhook(mgr, opts...); err != nil {
-				log.Error(err, "Unable to create webhook")
-				return err
-			}
-		} else {
-			if err := noop.SetupWebhook(mgr, cb.JobType); err != nil {
-				log.Error(err, "Unable to create noop webhook")
-				return err
+
+		if isFrameworkEnabled(cfg, name) {
+			if _, err := mgr.GetRESTMapper().RESTMapping(cb.GVK.GroupKind(), cb.GVK.Version); err != nil {
+				// TODO: If the below PR is released, we need to change a way to check if the GVK is registered.
+				// REF: https://github.com/kubernetes-sigs/controller-runtime/pull/2425
+				// if !meta.IsNoMatchError(err) {
+				//   return err
+				// }
+				// log.Info("No matching API server for job framework, skip to create controller and webhook")
+				var NoMatchingErr *discovery.ErrGroupDiscoveryFailed
+				if !errors.As(err, &NoMatchingErr) {
+					return err
+				}
+				log.Info("No matching API server for job framework, skip to create controller and webhook")
+			} else {
+				if err = cb.NewReconciler(
+					mgr.GetClient(),
+					mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, constants.KueueName)),
+					opts...,
+				).SetupWithManager(mgr); err != nil {
+					log.Error(err, "Unable to create controller")
+					return err
+				}
+				if err = cb.SetupWebhook(mgr, opts...); err != nil {
+					log.Error(err, "Unable to create webhook")
+					return err
+				}
+				return nil
 			}
 		}
+		if err := noop.SetupWebhook(mgr, cb.JobType); err != nil {
```

Comment:
Should think not be under else?

Reply:
In the current behavior, the `noop.SetupWebhook` is run even when `isFrameworkEnabled==true && crds.Has==false`.
So I think we need to run here even when `isFrameworkEnabled==true && errors.As(err, &NoMatchingErr)==true`. 

WDYT?